treessence/src/test/java/fr/bipi/treessence/file/FileLoggerTreeTest.kt

```diff
@@ -0,0 +1,396 @@
+package fr.bipi.treessence.file
+
+import android.content.Context
+import android.util.Log
+import fr.bipi.treessence.common.formatter.LogcatFormatter
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.unmockkAll
+import junit.framework.TestCase
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withContext
+import org.amshove.kluent.internal.assertEquals
+import org.hamcrest.CoreMatchers
+import org.hamcrest.MatcherAssert
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.jupiter.api.io.TempDir
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import timber.log.Timber
+import java.io.BufferedReader
+import java.io.File
+import java.io.FileReader
+import java.nio.file.Path
+import java.util.concurrent.atomic.AtomicInteger
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(Parameterized::class)
+class FileLoggerTreeTest(
+    private val coroutineScope: CoroutineScope?
+) {
+
+    @TempDir
+    private val tempDir: Path? = null
+
+    @MockK
+    private lateinit var context: Context
+
+    @MockK
+    private lateinit var file: File
```

Comment:
It doesn't look like you need to mock `file`. Instead, use local `val file` or `var file` in the tests as needed, and remove `file` from `setupDefaultMocks()`.